core/update.php

```diff
@@ -300,6 +300,10 @@ function update_info_page() {
   if (db_table_exists('cache_update')) {
     cache('update')->flush();
   }
+  // Flush the theme cache so we can render this page correctly if themes have
+  // been updated.
+  system_rebuild_theme_data();
```

Comment:
To solve the problem of additional or changed variables being passed to theme functions and templates, I don't think this line is necessary, which scans the file system for new themes.

This might solve _other_ potential problems but unless we have a situation in which we know it's solving a problem, we might be better off omitting this line to save unnecessary work when running updates.

```suggestion
```